pkg/lockfile/parse-conan-lock.go

```diff
@@ -0,0 +1,182 @@
+package lockfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type ConanReference struct {
+	Name            string
+	Version         string
+	Username        string
+	Channel         string
+	RecipeRevision  string
+	PackageID       string
+	PackageRevision string
+	TimeStamp       string
+}
+
+type ConanGraphNode struct {
+	Pref      string `json:"pref"`
+	Ref       string `json:"ref"`
+	Options   string `json:"options"`
+	PackageId string `json:"package_id"`
+	Prev      string `json:"prev"`
+	Path      string `json:"path"`
+	Context   string `json:"context"`
+}
+
+type ConanGraphLock struct {
+	Nodes map[string]ConanGraphNode `json:"nodes"`
+}
+
+type ConanLockFile struct {
+	Version string `json:"version"`
+	// conan v0.4- lockfiles use "graph_lock", "profile_host" and "profile_build"
+	GraphLock    ConanGraphLock `json:"graph_lock,omitempty"`
+	ProfileHost  string         `json:"profile_host,omitempty"`
+	ProfileBuild string         `json:"profile_build,omitempty"`
+	// conan v0.5+ lockfiles use "requires", "build_requires" and "python_requires"
+	Requires       []string `json:"requires,omitempty"`
+	BuildRequires  []string `json:"build_requires,omitempty"`
+	PythonRequires []string `json:"python_requires,omitempty"`
+}
+
+// TODO this is tentative and subject to change depending on the OSV schema
+const ConanEcosystem Ecosystem = "ConanCenter"
+
+func parseConanRenference(ref string) ConanReference {
+	// very flexible format name/version[@username[/channel]][#rrev][:pkgid[#prev]][%timestamp]
+	var reference ConanReference
+
+	parts := strings.SplitN(ref, "%", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		reference.TimeStamp = parts[1]
+	}
+
+	parts = strings.SplitN(ref, ":", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		parts = strings.SplitN(parts[1], "#", 2)
+		reference.PackageID = parts[0]
+		if len(parts) == 2 {
+			reference.PackageRevision = parts[1]
+		}
+	}
+
+	parts = strings.SplitN(ref, "#", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		reference.RecipeRevision = parts[1]
+	}
+
+	parts = strings.SplitN(ref, "@", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		UsernameChannel := parts[1]
+
+		parts = strings.SplitN(UsernameChannel, "/", 2)
+		reference.Username = parts[0]
+		if len(parts) == 2 {
+			reference.Channel = parts[1]
+		}
+	}
+
+	parts = strings.SplitN(ref, "/", 2)
+	if len(parts) == 2 {
+		reference.Name = parts[0]
+		reference.Version = parts[1]
+	} else {
+		reference.Version = ref
+	}
+	return reference
+}
+
+func parseConanV1Lock(lockfile ConanLockFile) []PackageDetails {
+	var reference ConanReference
+	packages := make([]PackageDetails, 0, len(lockfile.GraphLock.Nodes))
+
+	for _, node := range lockfile.GraphLock.Nodes {
+
```

Comment:
nit

```suggestion
	for _, node := range lockfile.GraphLock.Nodes {
```

Reply:
I don't see a change here :thinking:

Reply:
hmm I think GitHub bugged out - I've remade the change [here](https://github.com/google/osv-scanner/pull/59/files#r1064243726) across more lines so hopefully it'll stick